### ISSUE TITLE
Factor equation solving into its own judgment form

### ISSUE THREAD

**Issue by nrnrnr** [rogerburtonpatel/vml#24]
The evaluation rules for the solver all follow the same pattern, and there is so much clutter that I find it hard to follow what's going on.  I think it's time to try an auxiliary solver judgment form, with its own rules.  This sort of refactoring plays the same role in the theory that introducing a helper function would play in the code.  In fact, it may be easier to sketch the code first and then retrofit a solver function to a new judgment form.

**Comment by nrnrnr** [comment]
Note relation to #18.  This is a prerequisite.

**Comment by rogerburtonpatel** [comment]
Things look good here. We have multiple flavors of Verse that use this now.
